### ISSUE TITLE
GraphQL Schema Auto Fetch does not interpolate variables.

### ISSUE THREAD

**Issue by jared-mackey** [a85/POSTMan-Chrome-Extension#10806]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues and I know that duplicates will be closed

### Describe the Issue

When using a variable such as `{{host}}` as the URL for the GraphQL endpoint the request to auto fetch fails. In the console there is an error logged.

```
POST http://{{host}}/graphql
Error: getaddrinfo ENOTFOUND {{host}}

Request Headers
Authorization: Bearer {{accessToken}}
Content-Type: application/json
User-Agent: PostmanRuntime/7.29.0
Accept: */*
Host: {{host}}
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
Request Body
```

It seems none of the variables are interpolating properly.

### Steps To Reproduce

1. Create a collection
2. Set a collection variable named `host` and set it to `localhost:8080`
3. Create a new request
4. Set it as `POST` and put the url as `{{host}}/graphql`
5. Change the body of the request to GraphQL and trigger the auto-fetch.
6. Open the console and see the error showing it not interpolating the variables

### Operating System

Arch Linux

### Postman Version

9.15.8

### Postman Platform

Postman App

**Comment by giridharvc7** [comment]
Thanks for reporting this. 
We are tracking this as a bug. 

Environment variables should resolve and work with Grapql introspection, you can temporarily use this to unblock yourself if that is possible

**Comment by cupcakearmy** [comment]
Same here :)

**Comment by giridharvc7** [comment]
This issue is fixed with 9.18.2. Feel free to reopen this issue in case you see it

**Comment by Shereef** [comment]
![image](https://user-images.githubusercontent.com/356763/168644643-02e663a6-55dc-49c9-9998-75215e166e02.png)
Thanks so much!
I have tested this on Mac and it works!

**Comment by bo55vxr** [comment]
@giridharvc7 - I use variables for the Authorization token but these do not seem working as expected.

For the scenario below, the Authorization Header is NOT sent to the API and the Schema Fetch fails
![image](https://user-images.githubusercontent.com/1264998/169270144-e58be0c4-b5bd-49f4-a3d9-9ec25e1f2ada.png)

However, if you send an explicit value, then the Authorization header is sent to the API and the schema fetch succeeds
![image](https://user-images.githubusercontent.com/1264998/169270961-f5ba0dd9-caa4-408b-a5dc-b9fa64585c21.png)

**Comment by riya-saini** [comment]
@bo55vxr it seems to be working with `9.18.2`. Can you share the app version you're on and update if it's below `9.18.2`? You can also check the postman console to verify if the header is being sent correctly or not.

**Comment by bo55vxr** [comment]
> @bo55vxr it seems to be working with `9.18.2`. Can you share the app version you're on and update if it's below `9.18.2`? You can also check the postman console to verify if the header is being sent correctly or not.

I did explicitly do the update when this issue was resolved ... however, it seems I have transported to the future as it appears that I have 9.**19**.2
![image](https://user-images.githubusercontent.com/1264998/169279914-653a91e6-5a19-460f-a09d-2292b84a3b78.png)

POSTMAN Console shows Authorization Header is NOT being sent

![image](https://user-images.githubusercontent.com/1264998/169280188-7f452f2c-3a1b-4e25-a48e-fc71c6ad010e.png)

Changed the token to an explicit value and this time it DOES get sent 
![image](https://user-images.githubusercontent.com/1264998/169284026-a6f1af77-87c9-4c35-a057-23e73d82a54b.png)

**Comment by bo55vxr** [comment]
I would open as a new issue, but it appears that I am not able to

**Comment by giridharvc7** [comment]
Reopened this issue, thanks for reporting this. We've found the root cause and we'll push out a fix soon.

**Comment by joaodcp** [comment]
Any updates on this? I'm on Postman 9.29.0 and GraphQL Auto-Fetch request headers are not set to collection variables - if it is a collection variable, the header value will just be "{{variable}}".

**Comment by giridharvc7** [comment]
Can you confirm the Agent you are using? 
Agent information can be found here in case you are using the web app. 

<img width="390" alt="Screenshot 2022-08-22 at 4 46 46 PM" src="https://user-images.githubusercontent.com/1040583/185908715-3b43b0bb-8cfc-4eca-a102-7eb30c805ba1.png">

Can you also share some logs for us to debug?

**Comment by joaodcp** [comment]
@giridharvc7 I'm working locally, not signed in. Could that be the issue? If so, why?
Environment variables are working fine.

This is the only log I get from Auto-Fetch.
```
[requester][error]["SyncRequestService: This request is not allowed from scratchpad"]

[requester][warn]["GraphqlSchemaService~fetchAllSchemaMetaInfo: Could not refresh schema list",{"error":{"message":"SyncRequestService\\x7erequest: This request not allowed from scratchpad"}}]

[requester][info]["GraphqlIntrospector: Cache miss. Refreshing schema."]
```

**Comment by joaodcp** [comment]
Yup, seems to be the issue: collection variables work just fine on Workspaces. Why is that though? I couldn't find any documentation referring to that. Maybe you should add a warning. Or maybe this is unwanted behavior?

**Comment by SterlingChin** [comment]
Postman's new GraphQL client is currently in [open beta](https://blog.postman.com/announcing-postmans-new-graphql-client/). The new client should resolve this issue with variables.

**Comment by moinologics** [comment]
facing same issue in postman v10.20 as well.

**Comment by DSchau** [comment]
@moinologics can you supply a video or other reproduction if you're able?

I just tried the reproduction steps myself and was not able to reproduce.

Also: if folks haven't already tried, would recommend using the native GraphQL request type (in a new collection, create a new request of type GraphQL) instead of the HTTP POST with GraphQL body approach. Better support, a nice explorer view, and a more tailored experience for GraphQL users. I was _also_ not able to reproduce it there, as I checked both spots.

## HTTP Request View
<img width="1092" alt="Screenshot 2023-11-29 at 21 28 34" src="https://github.com/postmanlabs/postman-app-support/assets/3924690/00cf91f9-dc04-49f4-be69-90da0af76d44">

## GraphQL Request View

<img width="1091" alt="Screenshot 2023-11-29 at 21 25 45" src="https://github.com/postmanlabs/postman-app-support/assets/3924690/855850be-f72a-405b-ba35-d5ed304810d4">

**Comment by moinologics** [comment]
actually sometime it seems to work but sometimes it not. i will try to record a clip if possible and attach here.

**Comment by moinologics** [comment]
here is the video
https://github.com/postmanlabs/postman-app-support/assets/48400531/5548ad02-4f56-458d-946e-468739746522

**Comment by Wryte** [comment]
> I just tried the reproduction steps myself and was not able to reproduce.

Like @moinologics, I'm only having issues with the interpolation in headers during introspection. His was the authorization header but I'm using the Authorization tab and basic auth. Both the username and password are sent through as their variable names for the introspection query only instead of values when I print them on the server. When I make the actual request they are interpolated fine.

This was all in the GraphQL request view, I don't have the same issue in the http view. There I was using basic auth coming from a collection fed by env variables.

(Postman version 10.21.8)

**Comment by joaodcp** [comment]
I'm on 10.21.9 and having the same issue with the new GraphQL client. I'm working with an API whose tokens have a relatively short lifetime and I have a script executing before the query to refresh the token.
To get the schema obtained via introspection back I have to run a request, close the tab and reopen it again.
If I don't it just says the introspection request returned a 401 status code, while I can get other queries (written manually) from the API which indicates that it is interpolating the collection variable for other subsequent requests but not for the introspection request.

I'm also using the Authorization tab with a collection variable.

**Comment by apoorv-2204** [comment]
Same issue
